gcs/src/test/scala/org/broadinstitute/monster/storage/gcs/GcsApiSpec.scala

```diff
@@ -198,4 +219,203 @@ class GcsApiSpec extends FlatSpec with Matchers {
       }
       .unsafeRunSync()
   }
+
+  // statObject
+  it should "return true if a GCS object exists" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.GET
+      req.uri shouldBe statObjectURI
+      Resource.pure(Response[IO](body = Stream.emits("{}".getBytes())))
+    })
+
+    api
+      .statObject(bucket, path)
+      .unsafeRunSync() shouldBe (true -> None)
+  }
+
+  it should "return false if a GCS object does not exist" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.GET
+      req.uri shouldBe statObjectURI
+      Resource.pure(Response[IO](status = Status.NotFound))
+    })
+
+    api
+      .statObject(bucket, path)
+      .unsafeRunSync() shouldBe (false -> None)
+  }
+
+  it should "return the md5 of an existing object" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.GET
+      req.uri shouldBe statObjectURI
+      Resource.pure(
+        Response[IO](
+          body = Stream.emits(s"""{"${GcsApi.ObjectMd5Key}": "$theMd5"}""".getBytes())
+        )
+      )
+    })
+
+    api
+      .statObject(bucket, path)
+      .unsafeRunSync() shouldBe (true -> Some(theMd5))
+  }
+
+  // createObject
+  it should "create a GCS object" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.POST
+      req.uri shouldBe createObjectURI
+
+      Resource.pure(
+        Response[IO](
+          status = Status.Ok
+        )
+      )
+    })
+
+    api
+      .createObject(
+        bucket,
+        path,
+        `Content-Type`(MediaType.`text/event-stream`),
+        Some(theMd5),
+        bodyStream
+      )
+      .unsafeRunSync()
+  }
+
+  // deleteObject
+  it should "delete a GCS object" in {
```

Comment:
We should add another test case for deleting an object that doesn't exist.

The current behavior of the method would be to fail in that case, BUT that can be annoying. We could mirror the way most filesystem libraries work and change the method to return `IO[Boolean]` instead of `IO[Unit]` (as you suggested earlier). Then we can update the logic to return `true` when the file existed and was deleted, and `false` when it didn't exist.

Up to you if you want to do the boolean change in this PR, but we should add the 2nd test case for sure.